### PR TITLE
Update irradiance.klucher docs

### LINKED ISSUE #1405: Klucher diffuse sky docstring inconsistency
The equation for F' should contain DHI rather than Id0.

**Versions:**
 - ``pvlib.__version__``: 0.9 and earlier

### LINKED ISSUE #2183: Missing Descriptions for Formula Components in `haydavies`
The formula components for the [`haydavies`](https://pvlib-python.readthedocs.io/en/stable/reference/generated/pvlib.irradiance.haydavies.html#pvlib.irradiance.haydavies) function seems to not be properly defined. The current formula is:

$I_{d} = DHI \left( A R_b + (1 - A) \left(\frac{1 + \cos \beta}{2}\right) \right)$

However, the documentation does not provide definitions or explanations for the components $A$ and $R_b$. While $DHI$ and $\beta$ are relatively straightforward, I feel it would be beneficial to include their descriptions for completeness. 

Additionally, [`isotropic`](https://pvlib-python.readthedocs.io/en/stable/reference/generated/pvlib.irradiance.isotropic.html), [`klutcher`](https://pvlib-python.readthedocs.io/en/stable/reference/generated/pvlib.irradiance.klucher.html) and [`reindl`](https://pvlib-python.readthedocs.io/en/stable/reference/generated/pvlib.irradiance.reindl.html)   also lack description of formula components.

#### Request

1. Update the documentation for the [`haydavies`](https://pvlib-python.readthedocs.io/en/stable/reference/generated/pvlib.irradiance.haydavies.html#pvlib.irradiance.haydavies) function to include descriptions for the following components:
   - $A$
   - $R_b$
   - $DHI$
   - $\beta$

2. Update the documentation for the [`isotropic`](https://pvlib-python.readthedocs.io/en/stable/reference/generated/pvlib.irradiance.isotropic.html), [`klutcher`](https://pvlib-python.readthedocs.io/en/stable/reference/generated/pvlib.irradiance.klucher.html) and [`reindl`](https://pvlib-python.readthedocs.io/en/stable/reference/generated/pvlib.irradiance.reindl.html) (shall different issues be created for each function?)

I think that providing these definitions will improve clarity.

Thank you!

### PR DESCRIPTION
**Pull request by RDaxini**
- [X] Partially closes #2183, closes #1405
 - [X] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - [x] Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/main/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
 - [X] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
 - [x] Pull request is nearly complete and ready for detailed review.
 - [x] Maintainer: Appropriate GitHub Labels (including `remote-data`) and Milestone are assigned to the Pull Request and linked Issue.

[Docs here](https://pvlib-python--2192.org.readthedocs.build/en/2192/reference/generated/pvlib.irradiance.klucher.html#pvlib.irradiance.klucher)

### CODE REVIEW

**Comment by RDaxini** [comment]
I am trying to figure out what theta is in the original `pvlib.irradiance.kulcher` equation...
It looks like psi from the original publication ([see page 4](https://books.google.co.uk/books?hl=en&lr=&id=j95CAQAAIAAJ&oi=fnd&pg=PA2&dq=Evaluation+of+models+to+predict+insolation+on+tilted+surfaces+pdf&ots=a06kU6ms54&sig=sUsn1p6_xrLaNOlhh1OlnTlBjr0#v=onepage&q=modulating%20function&f=false)) and that is defined as the "angle between the sun direction and normal direction". That sounds like the solar zenith angle to me...? But in the original docs there is a separate theta_z for the solar zenith angle (90-solar elevation in the original publication) 
Or have I really mixed my angles up here? :S

**Comment by RDaxini** [comment]
Marked as draft until #2191 has been addressed to avoid duplicate reviews due to similarities between the changes made in these PRs

File: docs/sphinx/source/whatsnew/v0.11.1.rst [hunk 1]
```diff
@@ -74,7 +74,8 @@ Documentation
 * Removed unused "times" input from dni_et() function (:issue:`2105`)
 * Updated :py:func:`pvlib.irradiance.haydavies` to include equation variable
   definitions and a new "notes" section (:issue:`2183`, :pull:`2191`)
-
+* Updated :py:func:`pvlib.irradiance.klucher` to include equation variable
+  definitions and a new "notes" section (:issue:`2183`, :pull:`2192`)
```
[the remark this answers is not in the document]
Comment (kandersolar) [reply docs/sphinx/source/whatsnew/v0.11.1.rst:78]: I'd merge them

File: pvlib/irradiance.py [hunk 2]
```diff
@@ -630,66 +630,75 @@ def klucher(surface_tilt, surface_azimuth, dhi, ghi, solar_zenith,
...
+        The sky diffuse component of the solar radiation. [Wm⁻²]
+
+    Notes
+    -----
+    The Klucher (1979) model determines the diffuse irradiance
+    from the sky (ground reflected irradiance is not included in this
+    algorithm) on a tilted surface using the surface tilt angle, surface
+    azimuth angle, diffuse horizontal irradiance, direct normal irradiance,
+    global horizontal irradiance, extraterrestrial irradiance, sun zenith
+    angle, and sun azimuth angle. The expression for the diffuse irradiance,
```
Comment (kandersolar) [inline pvlib/irradiance.py:674]: I know this text is not new, but it doesn't seem correct to me.  Where are DNI and extraterrestrial DNI used?  I think this paragraph could use a revision.

Comment (AdamRJensen) [reply pvlib/irradiance.py:674]: Agreed

Comment (cwhanse) [reply pvlib/irradiance.py:674]: Agree also. That text was in the original import from PVLib for Matlab, where it was a cut/paste without edit from some other sky diffuse function (probably perez).

File: pvlib/irradiance.py [hunk 3]
```diff
@@ -630,66 +630,74 @@ def klucher(surface_tilt, surface_azimuth, dhi, ghi, solar_zenith,
...
-
-       F' = 1 - (I_{d0} / GHI)^2
-
-    Klucher's 1979 model determines the diffuse irradiance from the sky
-    (ground reflected irradiance is not included in this algorithm) on a
-    tilted surface using the surface tilt angle, surface azimuth angle,
-    diffuse horizontal irradiance, direct normal irradiance, global
-    horizontal irradiance, extraterrestrial irradiance, sun zenith
-    angle, and sun azimuth angle.
+    using the Klucher (1979) model [1]_ [2]_.
```
Comment (kandersolar) [inline pvlib/irradiance.py:633]: We should not put references in the first sentence of a docstring.  See #911.

Can you fix it with `haydavies` too?  Sorry I overlooked that in the review of #2191.

**Review by kandersolar — APPROVED** [review]

File: pvlib/irradiance.py (hunk 3 above)
Comment (cwhanse) [reply pvlib/irradiance.py:633]: > We should not put references in the first sentence of a docstring. See #911.

But they seem to be working as intended...maybe something has improved in sphinx.

Comment (RDaxini) [reply pvlib/irradiance.py]: > We should not put references in the first sentence of a docstring. See #911.

Noted. I'll add that instruction into the new contributing page style guide references section (in progress)

> Can you fix it with `haydavies` too? Sorry I overlooked that in the review of #2191.
done in the last commit, is that okay in this same PR?

**Comment by cwhanse** [comment]
@RDaxini please mark this PR as also closing #1405

**Review by cwhanse — APPROVED** [review]

**Comment by RDaxini** [comment]
> @RDaxini please mark this PR as also closing #1405
> 
> NVM I could do it so I did.

Thanks! Had no idea that issue existed. I have marked #2191 and #2193 as "related to" as well since it was noted in the discussion that:
> Seems like several of those items apply to the other sky diffuse model docstrings as well.

How should I resolve this comment:
> To discuss: remove "must be >=0", or change to "should be >=0". The code doesn't require this restriction, although the model certainly isn't valid outside that range. My view: model constraints on input (should be's) would be better placed in the description or in Notes, and code restrictions (must be's) in the parameter descriptions.

My view: if we keep the (recommended) constraint written in the docs, +1 to changing to "should be" and +0.75 on moving to notes/descriptions.

**Comment by cwhanse** [comment]
> How should I resolve this comment:

Let's address it in a follow-on PR. I agree with your proposed rule (model constraints only in the parameter description, other restrictions in the Notes). But I think text like this is found in many places.

The original intent was to inform users of restrictions that would guard against invalid results. That intent wasn't applied consistently, and in places, parameter ranges for model validation were substituted. If we keep ranges in the Notes we need to agree on their purpose.

**Comment by RDaxini** [comment]
@cwhanse okay, I understand that better now. Thanks for explaining.
So in that case, I think this PR is read to go:)

**Comment by kandersolar** [comment]
Thanks @RDaxini!

**Pull request merged**